src/main/scala/Domain/Counter.scala

```diff
@@ -52,21 +56,15 @@ object CounterProjection {
 
   type Data = TreeMap[AggregateId, Int]
 
-  def emptyCounterProjection = Projection.empty[Data](new TreeMap())
-
-  implicit object CounterHandler extends Projection.Handler[Counter.Event, Data] {
-
-    import Counter._
-
-    def hashPrefix = "Counter"
-
-    def handle(id: AggregateId, e: Event, d: Data) = e match {
-      case Created(id) =>
-        println("created " + id); d
-      case Incremented =>
-        println("+1"); d.updated(id, d.get(id).fold(1)(_ + 1))
-      case Decremented => println("-1"); d.updated(id, d.get(id).fold(-1)(_ - 1))
-    }
-  }
+  def emptyCounterProjection = Projection.empty[Data](TreeMap.empty, List(
+    (Counter.tag, createEventDataConsumer( (d: Data, t: Tag, id: AggregateId, v: Int, e: Counter.Event) => {
```

Comment:
TODO: create intermediate type here

extract serialisation to a level above DB adapter operations and leave DB only for storing / retrieving strings?